app/commands/user_track/generate_concepts_map_data.rb

```diff
@@ -0,0 +1,24 @@
+class UserTrack
+  class GenerateConceptsMapData
+    include Mandate
+
+    initialize_with :user_track
+
+    def call
+      Track::DetermineConceptsLayout.(user_track.track).tap { |data| data[:status] = concepts_status }
```

Comment:
I think we probably want to do this separately at the controller level. 

We'll be caching the `Track::DetermineConceptsLayout.(user_track.track)` data in the database (or maybe in the cache) so we'll want to  do the merging elsewhere I think. 

By having them as separate data sets, we can then be conditional in the controller about whether we merge it at all, or whether we show the track without the user data in the "logged out" version.

Reply:
Ok, I can simplify this to just apply the status to he concepts. Then, as you said, it can be joined in a controller